storage/entities/account.js

```diff
@@ -462,6 +497,200 @@ class Account extends BaseEntity {
+			this.SQLs.resetUnconfirmedState,
+			{},
+			{},
+			tx
+		);
+	}
+
+	/**
+	 * Clear data in memory tables:
+	 * - mem_round
+	 * - mem_accounts2delegates
+	 * - mem_accounts2u_delegates
+	 * - mem_accounts2multisignatures
+	 * - mem_accounts2u_multisignatures
+	 *
+	 * @param {Object} tx - DB transaction object
+	 * @returns {Promise}
+	 */
+	resetMemTables(tx) {
+		return this.adapter.executeFile(this.SQLs.resetMemTables, {}, {}, tx);
+	}
+
+	/**
+	 * Increment a field value in mem_accounts.
+	 *
+	 * @param {filters.Account} [filters] - Filters to match the objects
+	 * @param {string} field - Name of the field to increment
+	 * @param {Number|string} value - Value to be incremented
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	increment(filters, field, value, tx) {
+		return this._updateField(filters, field, value, '+', tx);
+	}
+
+	/**
+	 * Decrement a field value in mem_accounts.
+	 *
+	 * @param {filters.Account} [filters] - Filters to match the objects
+	 * @param {string} field - Name of the field to increment
+	 * @param {Number|string} value - Value to be incremented
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	decrement(filters, field, value, tx) {
+		return this._updateField(filters, field, value, '-', tx);
+	}
+
+	/**
+	 * Create dependent record for the account
+	 *
+	 * @param {string} dependencyName - Name of the dependent table
+	 * @param {string} address - Address of the account
+	 * @param {string} dependentPublicKey - Dependent public id
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	createDependentRecord(dependencyName, address, dependentPublicKey, tx) {
+		return this._updateDependentRecord(
+			dependencyName,
+			address,
+			dependentPublicKey,
+			'+',
+			tx
+		);
+	}
+
+	/**
+	 * Delete dependent record for the account
+	 *
+	 * @param {string} dependencyName - Name of the dependent table
+	 * @param {string} address - Address of the account
+	 * @param {string} dependentPublicKey - Dependent public id
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	deleteDependentRecord(dependencyName, address, dependentPublicKey, tx) {
+		return this._updateDependentRecord(
+			dependencyName,
+			address,
+			dependentPublicKey,
+			'-',
+			tx
+		);
+	}
+
+	_updateDependentRecord(
+		dependencyName,
+		address,
+		dependentPublicKey,
+		mode = '+',
+		tx
+	) {
+		assert(
+			Object.keys(dependentFieldsTableMap).includes(dependencyName),
+			`Invalid dependency name "${dependencyName}" provided.`
+		);
+		const params = {
+			tableName: dependentFieldsTableMap[dependencyName],
+			accountId: address,
+			dependentId: dependentPublicKey,
+		};
+		const SQL =
+			mode === '+'
+				? this.SQLs.createDependentRecord
+				: this.SQLs.deleteDependentRecord;
+
+		return this.adapter.executeFile(
+			SQL,
+			params,
+			{ expectedResultCount: 0 },
+			tx
+		);
+	}
+
+	_updateField(filters, field, value, mode = '+', tx) {
```

Comment:
Default value for mode isn't required as this is an internal method and we have the control of its use

```suggestion
	_updateField(filters, field, value, mode, tx) {
```